### DISCUSSION TITLE
Surf in china

### DISCUSSION THREAD

**Discussion by mg1067** [sinamics/ztnet#159] [General]
Hi.. 
my question is. Can I surf in china with my selfhosted zerotier server ?
Do I need some special settings ?

Thanks in advanced.
michael

**Comment by sinamics** [comment]
Yes you probably can, but please provide more details about your specific goal so I can assist you.

**Reply by mg1067** [reply]
Hello.. 

I think you know that no vpn is working in china. So I was in china last two weeks and installed a selfhosted zerotier server at home.
But I can not surfing in china. I use my notebook and my android mobile phone. I ping my computer at home and this was running. 
Now I think it must be DNS Problems. My son stay in china this moment and can test it for me for the next time. 
I am not in business there only privat. My wife is coming from china. I like to use my apps like in europe. 

I have a Network range set. In DNS I dont know what I must do. Maybe IP 8.8.8.8 and DNS Name google.de ?? 
In Managed routes I have my network 192.168.100.0/24 via my Router Gateway 192.168.10.1

In my Android app I set Route all traffic through zeroTier and I activate Network DNS.
In App Setting I check "Allow mobile Data" and uncheck  "Disable IPv6".

Do I need something else ?

Greetings
michael

**Comment by sinamics** [comment]
> I think you know that no vpn is working in china.

I did not know that.  

### Step 1: Enable IPv4 Forwarding

1. Open the sysctl configuration file by editing `/etc/sysctl.conf`.
2. Find and uncomment the line `net.ipv4.ip_forward` to enable IP forwarding upon boot.
3. To enable IP forwarding immediately, run the following command:

```bash
sudo sysctl -w net.ipv4.ip_forward=1
```

### Step 2: Configure iptables for NAT and Forwarding
Replace `PHY_IFACE` and `ZT_IFACE` with your actual physical and ZeroTier interface names.
```bash
PHY_IFACE=eth0
ZT_IFACE=ztxxxxxxx

sudo iptables -t nat -A POSTROUTING -o $PHY_IFACE -j MASQUERADE
sudo iptables -A FORWARD -i $ZT_IFACE -o $PHY_IFACE -j ACCEPT
sudo iptables -A FORWARD -i $PHY_IFACE -o $ZT_IFACE -m state --state RELATED,ESTABLISHED -j ACCEPT
```

After setting up your iptables rules, you'll want to make sure they are persistent across reboots. You can achieve this by installing `iptables-persistent` , you will be asked if you want to save the iptable config.

```bash
sudo apt install iptables-persistent
sudo sh -c 'iptables-save > /etc/iptables/rules.v4'
```

### Step 3: Update ZeroTier Network Configuration

In your Ztnet network settings, add `0.0.0.0/0` via your ZeroTier Server node IP address. 

Example:
![image](https://github.com/sinamics/ztnet/assets/7771916/56c9a7fd-e9f3-43f2-ba71-806c03859afe)

After completing these steps, you'll be able to route your internet traffic through the server, thereby bypassing any IP-based restrictions and allowing you to browse the internet as if you're in China.

**Comment by mg1067** [comment]
Step 2:  I don't understand. On my server ? I have no zt interface. Only on Clients. Or I am wrong ? 
Step 3: I have there 192.168.195.0/24 via 192.168.10.1  First is my ZT Range and second my Real Router IP.

**Comment by sinamics** [comment]
> I don't understand. On my server ? I have no zt interface. Only on Clients. Or I am wrong ?

the client you want to route your traffic though must be connected to the zerotier network. When connected you will see a zerotier network interface starting with `zt`.  on the client, type `ip a` to see all the interfaces

> I have there 192.168.195.0/24 via 192.168.10.1 First is my ZT Range and second my Real Router IP.

That wont work. 
Set destination to `0.0.0.0/0`  via the zerotier ip address of the node you want to route traffic through.

**Reply by mg1067** [reply]
So I need only one client as a router. Right ?

**Reply by sinamics** [reply]
yes

**Comment by mg1067** [comment]
Ah... Now I understand... Thanks.. I will test it.. If I have more Questions I will ask you.

**Comment by mg1067** [comment]
Sorry.. Only One question... 
192.168.195.0/24 via LAN means ? IP 192.168.195.... or my real network 192.168.10.1 ??
What I must write as LAN ?

**Comment by sinamics** [comment]
I made a how-to in the docs. 
https://ztnet.network/usage/override_default_route

Does this help?
